lib/gooddata/lcm/actions/associate_clients.rb

```diff
@@ -46,7 +49,21 @@ class AssociateClients < BaseAction
       ]
 
       class << self
+        DELETE_MODES = %w(
+          none
+          delete_projects
+          delete_extra
+        )
         def call(params)
+          if params.delete_projects && !params.delete_extra
```

Comment:
We must convert the params to bools. `'false'` evaluates to `true` in a logical expression. There is also a ticket in JIRA for converting bool params automatically.